src/proof/proof_node_updater.cpp

```diff
@@ -172,21 +189,26 @@ void ProofNodeUpdater::processInternal(std::shared_ptr<ProofNode> pf,
         Assert(fa.size() >= args.size());
         fa.resize(fa.size() - args.size());
       }
-      runFinalize(cur, fa, resCache, resCacheNcWaiting, cfaMap);
+      // run update (marked as post-visit) to a fixed point
+      bool continueUpdate = true;
+      while (runUpdate(cur, fa, continueUpdate, false) && continueUpdate)
+      {
+        Trace("pf-process-debug") << "...updated proof." << std::endl;
+      }
+      if (!continueUpdate)
```

Comment:
We should not have this check. (In fact, in the current state, it makes it so that we never run finalize I believe).

Moreover, I don't think we should have a `continueUpdate` argument passed to the callback's shouldUpdatePost or updatePost at all, since they shouldnt have the authority on whether to run runFinalize.